client/cmd.go

```diff
@@ -55,3 +58,59 @@ func ValidateCmd(cmd *cobra.Command, args []string) error {
 
 	return cmd.Help()
 }
+
+// ReadGetCommandFlags returns an updated Context with fields set based on flags
+// defined in GetCommands. An error is returned if any flag query fails.
+func ReadGetCommandFlags(clientCtx Context, flagSet *pflag.FlagSet) (Context, error) {
+	if flagSet.Changed(flags.FlagHeight) {
+		height, err := flagSet.GetInt64(flags.FlagHeight)
+		if err != nil {
+			return clientCtx, err
+		}
+
+		clientCtx = clientCtx.WithHeight(height)
+	}
+
+	if flagSet.Changed(flags.FlagTrustNode) {
```

Comment:
Is the idea that flags which are read from config files (i.e. with viper) get boundin `clientCtx` earlier and here we're only looking for overrides?

I was thinking that we may need to pass a `*Viper` instance around in `clientCtx` and then call `BindFlagValues` here in this method, but pre-populating `clientCtx` with values from config files may be the better approach...

Reply:
Yes, correct.

When the root command is created and child (module) commands are added, they're given a `client.Context` ([ref](https://github.com/cosmos/cosmos-sdk/blob/master/simapp/cmd/simcli/main.go#L125-L130)). This `client.Context` may have fields set explicitly (as in the ref) and/or via config too, in addition to tests. The idea is we need to keep these values set and unchanged _unless_ explicitly overridden by the client, hence we use `flagSet.Changed`.